application/client/src/app/ui/views/toolbar/chart/chart/chart.ts

```diff
@@ -0,0 +1,411 @@
+                        'line',
+                        ScatterDataPoint[]
+                    >;
+                    dataset.stepped = entity.definition.stepped;
+                    dataset.borderWidth = entity.definition.borderWidth;
+                    dataset.tension = entity.definition.tension;
+                    dataset.pointRadius = entity.definition.pointRadius;
+                }
+            }
+        });
+        // this._updateYAxes(entities);
+        this._chart.update();
+        this._updateHasNoData();
+    }
+
+    private _removeRedundantDatasets(entities: ChartRequest[]) {
+        this._chart.data.datasets = this._chart.data.datasets.filter(
+            (
+                dataset: ChartDataset<
+                    keyof ChartTypeRegistry,
+                    (number | ScatterDataPoint | BubbleDataPoint | null)[]
+                >,
+            ) => {
+                return (
+                    entities.findIndex((entity: ChartRequest) => {
+                        return (
+                            entity.definition.filter === dataset.label && entity.definition.active
+                        );
+                    }) !== -1
+                );
+            },
+        );
+    }
+
+    private _initDatasets() {
+        this._chart.data.datasets = [];
+        this._session.search
+            .store()
+            .charts()
+            .getActive()
+            .forEach((request: ChartRequest, index: number) => {
+                // const uuid: string = request.uuid();
+                this._chart.data.datasets[index] = {
+                    label: request.definition.filter,
+                    data: [],
+                    backgroundColor: request.definition.color,
+                    borderColor: request.definition.color,
+                    stepped: request.definition.stepped,
+                    spanGaps: true,
+                    borderWidth: request.definition.borderWidth,
+                    tension: request.definition.tension,
+                    pointRadius: request.definition.pointRadius,
+                    // yAxisID: uuid,
+                };
+                // this._showYAxis(request);
+            });
+    }
+
+    private _initData() {
+        if (this._session.search.store().charts().getActive().length === 0) {
+            this._updateHasNoData();
+            return;
+        }
+        this._labelState.loading = true;
+        this._chart.data.labels = [];
+        for (const [line, values] of this._session.search.values.get()) {
+            values.forEach((sValue: string, id: number) => {
+                const value: number = parseInt(sValue);
+                const scaleX = (this._chart.config.options as any).scales['x'];
+                if (line < scaleX.min) {
+                    scaleX.min = line;
+                } else if (line > scaleX.max) {
+                    scaleX.max = line;
+                }
```

Comment:
is condition `line === scaleX`  missed?

Reply:
No, if `line === scaleX.min` or `line  === scaleX.max`, we can just skip. This is because we don't need to update `min` or `max` in that case. What we do here is update the `min` and `max` of the scale so if for example: `max = 10`, 'there is no point to reassign `max` to `10` again.